app/support/welcome-directs.ts

```diff
@@ -0,0 +1,189 @@
+import { readFile } from 'fs/promises';
+
+import createDebug from 'debug';
+import { loadAll } from 'js-yaml';
+import { z } from 'zod';
+import { DateTime, Duration } from 'luxon';
+import { fromError, isZodErrorLike } from 'zod-validation-error';
+
+import { dbAdapter, User, Job, Post, Comment, JobManager } from '../models';
+
+import { currentConfig } from './app-async-context';
+import { UUID } from './types';
+import { isNoEntryError } from './is-no-entry';
+
+export const WELCOME_DIRECT = 'WELCOME_DIRECT';
+
+export function initHandlers(jobManager: JobManager) {
+  jobManager.on(WELCOME_DIRECT, jobHandler);
+}
+
+const debugLog = createDebug('freefeed:welcomeDirects');
+const errorLog = createDebug('freefeed:welcomeDirects:errors');
+
+export type JobPayload = {
+  id: string;
+  senderId: UUID;
+  userId: UUID;
+  body: string;
+  comment?: string;
+  criterion?: { type: string };
+};
+
+const entrySchema = z.object({
+  id: z.string(),
+  delay: z.string().duration().default('PT0S'),
+  skipDays: z.number().int().positive().optional(),
+  criterion: z
+    .object({
+      type: z.string(),
+      // ...some other arguments
+    })
+    .optional(),
+  body: z.string(),
+  comment: z.string().optional(),
+});
+
+export async function scheduleWelcomeDirects(user: User): Promise<boolean> {
+  const { senderAccount, scheduleFile } = currentConfig().welcomeDirects;
+
+  if (!scheduleFile) {
+    // No schedule file is configured, so nothing to do
+    return false;
+  }
+
+  const sender = await dbAdapter.getUserByUsername(senderAccount);
+
+  if (!sender) {
+    errorLog(`User "${senderAccount}" does not exist`);
+    throw new Error(`User "${senderAccount}" does not exist`);
+  }
+
+  let entries: z.infer<typeof entrySchema>[];
```

Comment:
you use `z.infer<typeof entrySchema>` couple of times. makes sense to extract it into the peroper `type`